### PR TITLE
Always return response data in a value field

### PR DESCRIPTION
**Pull request by shs96c**
This change is [Reviewable](https://reviewable.io/reviews/w3c/webdriver/345)

### CODE REVIEW

**Comment by andreastt** [comment]
For the benefit of posterity, there is relevant discussion about this PR in https://lists.w3.org/Archives/Public/public-browser-tools-testing/2016OctDec/0001.html.

(At some point we should also move discussions onto here.)

File: webdriver-spec.html [hunk 1]
```diff
@@ -734,8 +734,9 @@
   to the description of <var>status</var> in the <a>status code registry</a>.
 
  <li><p>If <var>data</var> is not null, let <var>response</var>’s
-  <a>body</a> be the result of <a>serializing as JSON</a>
-  with <var>data</var> as the argument.
+  <a>body</a> be a JSON object with a top-level property
```
Comment (andreastt) [inline webdriver-spec.html]: `JSON <a>Object</a>`

Comment (andreastt) [inline webdriver-spec.html]: I also think saying “top-level property” is redundant. You can just say `… JSON <a>Object</a> with a key <code>value</code> set to the <a>JSON serialisation</a> of <var>data</var>`.

File: webdriver-spec.html [hunk 2]
```diff
@@ -734,8 +734,9 @@
...
+  of <var>value</var>, the value of which is the result
```
Comment (andreastt) [inline webdriver-spec.html]: This should be `<code>value</code>` as it’s not a variable in the algorithm, but a key on the JSON Object.

File: webdriver-spec.html [hunk 3]
```diff
@@ -2248,10 +2249,8 @@
   of the <a>current top-level browsing context</a>’s
   <a>active document</a>’s <a>document URL</a>.
 
- <li><p>Let <var>body</var> be a JSON Object
-  with the "<code>value</code>" property set to <var>url</var>.
-
- <li><p>Return <a>success</a> with data <var>body</var>.</p></li>
+ <li><p>Return <a>success</a> with data set
+  to <var>url</var>.</p></li>
```
Comment (andreastt) [inline webdriver-spec.html]: `with data <var>url</var>` to be consistent with the other algorithms.

File: webdriver-spec.html [hunk 4]
```diff
@@ -2441,12 +2440,8 @@
...
  context</a>'s <a>active document</a>.
 
- <li><p>Let <var>body</var> be a new JSON Object.
-
- <li><p>Set the property "<code>value</code>" of <var>body</var>
-  to the value of <var>title</var>.
-
- <li><p>Return <a>success</a> with data <var>body</var>.
+ <li><p>Return <a>success</a> with data being the value
+ of <var>title</var>.
```
Comment (andreastt) [inline webdriver-spec.html]: Same here. Please let us just say “Return success with data _title_” throughout this patch.

**Comment by andreastt** [comment]
Also, I don’t believe you are handling the return value from finding single- and multiple elements correctly. This requires a change too.

**Comment by shs96c** [comment]
Updated the patch. 

If we were handling finding elements correctly before, we will be doing so now. The change to how we encode responses should mean that where we used to return the raw element or array of elements we now return a JSON object with the value set properly.

File: webdriver-spec.html (hunk 1 above)
Comment (shs96c) [inline webdriver-spec.html]: Done.

File: webdriver-spec.html (hunk 2 above)
Comment (shs96c) [inline webdriver-spec.html]: Fixed by the above

File: webdriver-spec.html (hunk 3 above)
Comment (shs96c) [inline webdriver-spec.html]: Done.

File: webdriver-spec.html (hunk 4 above)
Comment (shs96c) [inline webdriver-spec.html]: Done and done.

**Comment by andreastt** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/webdriver/345)*

**Pull request merged**